File: src/main/java/sample/commands/GUICommand.java
package sample.commands;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.SceneAntialiasing;
import javafx.scene.image.Image;
import javafx.stage.Stage;
import picocli.CommandLine;
import sample.controller.MainController;

import java.io.IOException;

@CommandLine.Command(name = "GUI", aliases = "", description = "Starts the CAViewer GUI")
public class GUICommand extends Application implements Runnable {
    private MainController controller;

    public static void main(String[] args) {
        launch(args);
    }

    @Override
    public void start(Stage stage) throws IOException {
        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("/main.fxml"));
        Scene scene = new Scene(fxmlLoader.load(),700, 650, true, SceneAntialiasing.DISABLED);
        scene.getStylesheets().add("/style.css");

        controller = fxmlLoader.getController();

        stage.setScene(scene);
        stage.setTitle("Cellular Automaton Viewer");
        stage.getIcons().add(new Image(getClass().getResourceAsStream("/icon/PulsarIcon.png")));
        stage.show();
    }

    @Override
    public void stop(){
        controller.onApplicationClosed();
    }

    @Override
    public void run() {
        launch();
    }
}

File: src/main/kotlin/application/commands/GUICommand.kt
package application.commands

import java.lang.Runnable
import application.controller.MainController
import javafx.application.Application
import kotlin.Throws
import java.io.IOException
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.scene.SceneAntialiasing
import javafx.scene.image.Image
import javafx.stage.Stage
import picocli.CommandLine

@CommandLine.Command(name = "GUI", aliases = [""], description = ["Starts the CAViewer GUI"])
class GUICommand : Application(), Runnable {
    private var controller: MainController? = null
    @Throws(IOException::class)
    override fun start(stage: Stage) {
        val fxmlLoader = FXMLLoader(javaClass.getResource("/main.fxml"))
        val scene = Scene(fxmlLoader.load(), 700.0, 650.0, true, SceneAntialiasing.DISABLED)
        scene.stylesheets.add("/style.css")
        controller = fxmlLoader.getController()

        stage.scene = scene
        stage.title = "Cellular Automaton Viewer"
        stage.icons.add(Image(javaClass.getResourceAsStream("/icon/PulsarIcon.png")))
        stage.show()
    }

    override fun stop() {
        controller!!.onApplicationClosed()
    }

    override fun run() {
        launch()
    }
}